internal/chain/create_key_book.go

```diff
@@ -14,77 +14,63 @@ type CreateKeyBook struct{}
 
 func (CreateKeyBook) Type() types.TxType { return types.TxCreateKeyBook }
 
-func checkCreateKeyBook(st *StateManager, tx *transactions.GenTransaction) ([]*protocol.SigSpec, *url.URL, error) {
+func (CreateKeyBook) Validate(st *StateManager, tx *transactions.GenTransaction) error {
 	if _, ok := st.Sponsor.(*state.AdiState); !ok {
-		return nil, nil, fmt.Errorf("invalid sponsor: want %v, got %v", types.ChainTypeAdi, st.Sponsor.Header().Type)
+		return fmt.Errorf("invalid sponsor: want %v, got %v", types.ChainTypeAdi, st.Sponsor.Header().Type)
```

Comment:
Would this be a more meaningful error if it output the type name instead of number?  For example, types.ChainTypeAdi.Name() and st.sponsor.Header().Type.Name()

Reply:
`%v` is smart and will call `.String()` on the value, so this will come out as `invalid sponsor: want ChainTypeAdi, got ChainTypeSigSpec` (for example).

Reply:
ahhh...